src/stan/lang/ast/node/array_expr_def.hpp

```diff
@@ -11,7 +11,8 @@ namespace stan {
                                array_expr_scope_() { }
 
     array_expr::array_expr(const std::vector<expression>& args)
-      : args_(args), type_(), has_var_(false), array_expr_scope_() { }
+      : args_(args), type_(bare_array_type(args.at(0).bare_type())),
```

Comment:
the parser uses 0-arg constructor and semantic actions are used to set member variables,
and the 1-arg constructor is not used anywhere in the code - should it be eliminated?

the doc on the 1-arg constructor advertises that 0-length vector is not allowed.
initializing type of 1-arg constructor is problematic for arrays which have both scalar int and scalar doubles - these should always be type-promoted to array of doubles.

agreed, should be indexing as `[0]` instead of `at(0)`

Reply:
convinced myself - removing 1-arg constructor.